File: test_mongodb.py
from pymongo import MongoClient
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# MongoDB connection string
mongo_uri = os.getenv('MONGODB_URI', 'mongodb://localhost:27017/college-sports-hub')

try:
    # Create a MongoDB client
    client = MongoClient(mongo_uri)
    
    # Test the connection
    client.admin.command('ping')
    print("Successfully connected to MongoDB!")
    
    # Get the database
    db = client['college-sports-hub']
    
    # List all collections
    collections = db.list_collection_names()
    print("\nCollections in database:")
    for collection in collections:
        print(f"- {collection}")
    
    # Close the connection
    client.close()
    
except Exception as e:
    print(f"Error connecting to MongoDB: {e}") 

File: test_mongodb.js
const mongoose = require('mongoose');
require('dotenv').config();

const mongoUri = process.env.MONGODB_URI || 'mongodb://localhost:27017/college-sports-hub';

async function testConnection() {
    try {
        // Connect to MongoDB
        await mongoose.connect(mongoUri, {
            useNewUrlParser: true,
            useUnifiedTopology: true
        });
        
        console.log('Successfully connected to MongoDB!');
        
        // List all collections
        const collections = await mongoose.connection.db.listCollections().toArray();
        console.log('\nCollections in database:');
        collections.forEach(collection => {
            console.log(`- ${collection.name}`);
        });
        
        // Close the connection
        await mongoose.connection.close();
        console.log('\nConnection closed.');
        
    } catch (error) {
        console.error('Error connecting to MongoDB:', error);
    }
}

testConnection(); 